hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-common/src/main/java/org/apache/hadoop/yarn/server/uam/UnmanagedApplicationManager.java

```diff
@@ -431,6 +431,7 @@ private void submitUnmanagedApp(ApplicationId appId) throws YarnException, IOExc
     context.setResource(resource);
     context.setAMContainerSpec(amContainer);
     if (applicationSubmissionContext != null) {
+      context.setApplicationName(applicationSubmissionContext.getApplicationName());
```

Comment:
This is a good idea, but I'm still a little worried. The initial decision not to assign  application name was to quickly differentiate whether the task originated from a different sub-cluster. If we were to assign a name, it might make it more challenging to discern if the task belongs to another sub-cluster.

Reply:
Yes, you are right. I just wanna search the application by name. But It is the quick way to find home sub cluster indeed. I will close this pr firstly until someone will be interested in this.